### ISSUE TITLE
StackOverflowError on "Next Turn"

### ISSUE THREAD

**Issue by mliulion** [yairm210/Unciv#11349]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Game Version

4.10.20

### Describe the bug

I tapped "Next Turn" and it crashed

I already had Refrigeration and Flight techs

Reloading the game works, but it doesn't fix the bug (tapping "Next Turn")

The err
```
**Platform:** Android
**Version:** 4.10.20 (Build 980)
**Rulesets:** [Civ V - Vanilla, Civ V - Gods & Kings]
**Last Screen:** `com.unciv.ui.screens.worldscreen.WorldScreen`

--------------------------------

Device Model: ASUS_I006D
API Level: 33
System Memory: 7258 MB
    Available (used by Kernel): 1741 MB
    System Low Memory state: false
    Java heap limit: 768 MB
    Java heap free: 5 MB

--------------------------------


**Message:**
```
java.lang.StackOverflowError: stack size 1039KB
	at java.util.HashMap.hash(HashMap.java:336)
	at java.util.HashMap.getNode(HashMap.java:566)
	at java.util.HashMap.containsKey(HashMap.java:592)
	at java.util.HashSet.contains(HashSet.java:204)
	at com.unciv.models.ruleset.tile.Terrain.matchesSingleFilter(Terrain.kt:154)
	at com.unciv.models.ruleset.tile.Terrain$matchesFilter$1.invoke(Terrain.kt:148)
	at com.unciv.models.ruleset.tile.Terrain$matchesFilter$1.invoke(Terrain.kt:148)
	at com.unciv.logic.MultiFilter.multiFilter(MultiFilter.kt:13)
	at com.unciv.logic.MultiFilter.multiFilter$default(MultiFilter.kt:4)
	at com.unciv.models.ruleset.tile.Terrain.matchesFilter(Terrain.kt:148)
	at com.unciv.models.ruleset.tile.TileImprovement.canBeBuiltOn(TileImprovement.kt:79)
	at com.unciv.models.ruleset.tile.TileImprovement.isAllowedOnFeature(TileImprovement.kt:92)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:109)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$canBeBuildOnThisUnbuildableTerrain(TileInfoImprovementFunctions.kt:120)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovementBeBuiltHere$core(TileInfoImprovementFunctions.kt:146)
	at com.unciv.logic.map.tile.TileInfoImprovementFunctions.canImprovement
```

### Steps to Reproduce

I tapped "Next Turn" and it crashed

### Screenshots

_No response_

### Link to save file

_No response_

### Operating System

Android

### Additional Information

_No response_

**Comment by mliulion** [comment]
More info:

![Screenshot_20240321-144354594](https://github.com/yairm210/Unciv/assets/22740034/d09a482c-2f51-4584-be2a-6e49e5e3919d)

![Screenshot_20240321-144407695](https://github.com/yairm210/Unciv/assets/22740034/084607fe-9729-4e9e-bc5a-7f4bf1b4def4)

![Screenshot_20240321-144419602](https://github.com/yairm210/Unciv/assets/22740034/75489594-ae7a-4934-a1ca-58ed13ac2e3d)

**Comment by 123darthvader321** [comment]
Same error
![Screenshot_2024-03-21-20-11-33-860_com unciv app](https://github.com/yairm210/Unciv/assets/164273698/4b676c5d-e0d3-4a21-afd5-c21727a30154)

**Comment by SeventhM** [comment]
Duplicate of https://github.com/yairm210/Unciv/issues/11332. Fixed in 4.10.21
